custom/icds_reports/static/js/angular-services/locations.service.js

```diff
@@ -109,5 +109,20 @@ window.angular.module('icdsApp').factory('locationsService', ['$http', '$locatio
                 (selectedLocations[level - 1] && selectedLocations[level - 1] !== 'all' && selectedLocations[level - 1].location_id !== 'all')
             );
         },
+        getLocationsForLevel: function(level, selectedLocations, locationsCache) {
+            if (level === 0) {
+                return locationsCache.root;
+            } else {
+                var selectedLocation = selectedLocations[level - 1];
+                if (!selectedLocation || selectedLocation.location_id === 'all') {
+                    return [];
+                }
+                return _.sortBy(
```

Comment:
The Modal and Filter controllers each had a copy of this function, but only one was sorting at the end. I suspect the other just never needed it since it didn't display a whole list anywhere. This is perhaps the largest assumption in the PR that has the opportunity to affect web dashboard (if some code is relying on an unsorted list instead of a sorted list). I tried to wrap my head around the 6-7 uses of it and confirm it seemed safe (and I think the deduplication of code is worth it) but feel free to push back if that doesn't sound reasonable.

Reply:
Thanks I did a code search and that isn't affected. It looks like that page doesn't funnel through this code pathway, but that code has copy/pasted a huge amount of logic from the `LocationsModalController`  (https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/static/js/directives/download/download.directive.js#L163-L296)

We aren't porting tabular reports to mobile so I'm not going to touch it, but it's one more piece of tech debt in the dashboard front end that's good to be aware of.

Reply:
It looks like whoever implemented that page didn't feel like creating the abstractions that would have been valuable at the time and now we're in this situation (which is already hopefully a bit better, but obviously still lots of room for improvement)

Reply:
I am not positive where this exact code path is, but we did run into an issue a few days ago where locations were sorted different on different parts of the dashboard, so if this resolves that, im all for it.

Reply:
Cool - if you have a link to the ticket I can check. At least possible this might fix it.